### ISSUE TITLE
[6.0] lv_style_anim_get_user_data_ptr: infinite recursion

### ISSUE THREAD

**Issue by rreilink** [lvgl/lvgl#1081]
Compiling lvgl gives several warnings:

```c
In file included from lvgl/src/lv_objx/lv_gauge.c:9:
In file included from lvgl/src/lv_objx/lv_gauge.h:29:
In file included from lvgl/src/lv_objx/../lv_core/lv_obj.h:24:
lvgl/src/lv_objx/../lv_core/lv_style.h:257:1: warning: all paths through this function will call itself [-Winfinite-recursion]
{
```

This warning is correct since indeed `lv_style_anim_get_user_data_ptr` calls itself:
```c
static inline lv_anim_user_data_t * lv_style_anim_get_user_data_ptr(lv_anim_t * a)
{
    return lv_style_anim_get_user_data_ptr(a);
}
```

**Comment by embeddedt** [comment]
(cc @kisvegabor)

Very strange. I assume that this function just hadn't been implemented yet.

**Comment by rreilink** [comment]
Should it read `return lv_anim_get_user_data_ptr(a);` ?

**Comment by embeddedt** [comment]
@rreilink I think so.

**Comment by kisvegabor** [comment]
> Should it read return lv_anim_get_user_data_ptr(a); ?

Yes, I fixed it here 6093053

Thank you for the report!
